### ISSUE TITLE
Configuration option to auto-join rooms.

### ISSUE THREAD

**Issue by DAMO238** [opsdroid/opsdroid#1772]
# Description
For connectors such as matrix (I don't actually use any others currently), if a user wants to use a bot in their room, the only way it is possible is if they contact the owner and get them to add their room into the config file. Obviously, this is not necessarily ideal, so I propose an option (that is false by default) that makes it so that, whenever the bot is invited to a room, it will accept the invitation automatically.

## Steps to Reproduce
Invite a bot into a matrix room.

## Expected Functionality
If the proposed option is set to true, the bot will join the room as soon as it gets the invite.

## Experienced Functionality
There is no option, so the bot will not join the room unless the room is added to the config.

## Versions
- **Opsdroid version: 0.21.0**
- **Python version: 3.7.3**
- **OS/Docker version: Debian 10**

## Additional Details
Loving how easy it is build a fast and reliable bot. Massive props to the maintainers and contributors!

**Comment by Cadair** [comment]
I agree this would be a good feature to have, but it's easy to write a skill which does this for an example see here: https://github.com/Cadair/skill-jitsi/blob/master/__init__.py#L249-L257

**Comment by DAMO238** [comment]
Thank you for that, that worked great! Is there anywhere I can read up on all the matrix specific events, since I couldn't find anything in the docs?

**Comment by Cadair** [comment]
The ones which are truly matrix specific are in this file: https://github.com/opsdroid/opsdroid/blob/master/opsdroid/connector/matrix/events.py all the others are in https://github.com/opsdroid/opsdroid/blob/master/opsdroid/events.py 

#1293 is the issue tracking improving the docs for connectors and events.
